Started working on this

0Xellos opened the pull request:
Macro parsing is expanded. Now if Event.Macro, then Event.args = (macro name, first non-whitespace char afterwards if it exists, optional arguments if they exist).
This allows roughly checking what the macro is applied to (if the next non-whitespace char is '{', applied to the next group, otherwise to that char, None if there's no such char before eoln). Implemented are checks if macros \left(, \right) and \dots{ are used. The checks are done using regexps - this allows greater flexibility, e.g. \left( needs to be forbidden but \dots{ is the only allowed \dots macro.

0Xellos committed:
expanded macro parsing

0Xellos committed:
added checkers for macros: dots, left(, right)

0Xellos committed:
too complex, needs to be done manually

0Xellos committed:
added checkers

Werkov commented:
Hi @0Xellos, long time no see. Thanks for your contribution.

I modified sample text as following
> při odvození Ciolkovského \dots rovnice 

and got
> Line 56:	Forbidden macro \dots r, use \dots{} instead.

This correctly assumes `r` may be the macro argument, however, this is not case for all sequences, there are nullary macros, for instance the `\dot` in particular.

Let me know if you plan continuing with this, I can merge the change as is.

(Also you may get in touch with current TeXnicians (@MeggyCal and @Auratons) to get genuine feedback from potential users. AFAICS, linter (sadly) is not used at the moment.)

0Xellos commented:
Yes, identifying the arguments of the macro is something the linter can't do now and most likely won't ever, it's just too complex.
That means the output can't mean more than "if the macro takes some arguments, here's a hint to what the next one will be". I'll add a check if this arg has a regex; an alternative is printing regexes, but that seems like too obscure output.

0Xellos committed:
added synchronized Event.Character

0Xellos committed:
improved output

0Xellos committed:
removed debug message

0Xellos committed:
regex None matches everything

0Xellos commented:
Added; also changed matching slightly. Now you should be able to forbid nullary macros with e.g. `('dot', None, None)` to get the error `Line X:	Forbidden macro \dot, use . instead.`
I'm closing this, branching and making separate pull requests because I didn't know other commits made to the same branch would also be included in the current pull request...

Werkov commented:
Yep, pull requests track a branch (which is a movable target). I'm thus not merging anything from this PR and looking forward for new PRs.
